iquip/apps/explorer.py

```diff
@@ -89,12 +113,20 @@ def openExperiment(self):
         experimentFileItem = self.explorerFrame.fileTree.currentItem()
         if experimentFileItem.childCount():
             return
-        experimentPath = experimentFileItem.text(0)
+        experimentPath = self.getFullPath(experimentFileItem)  # pylint: disable=unused-variable
+
+    def getFullPath(self, experimentFileItem: QTreeWidgetItem) -> str:
+        """Finds the full path of the file item and returns it.
+
+        Args:
+            experimentFileItem: The file item to get its full path.
+        """
+        path = experimentFileItem.text(0)
         while experimentFileItem.parent():
             experimentFileItem = experimentFileItem.parent()
-            experimentPath = posixpath.join(experimentFileItem.text(0), experimentPath)
-        experimentPath = posixpath.join(self.repositoryPath, experimentPath)
-
+            path = posixpath.join(experimentFileItem.text(0), path)
+        path = posixpath.join(self.repositoryPath, path)
```

Comment:
How about using a `list` and extending it, and calling `join()` at the end just once?
I think `join` operation is more expansive since it repeatedly creates and copies strings.

Reply:
Oh, it's better! I will apply it.